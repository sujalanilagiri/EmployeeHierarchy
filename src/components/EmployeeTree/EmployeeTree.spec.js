import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { shallow } from 'enzyme';
import EmployeeComponent from './EmployeeTree';

configure({ adapter: new Adapter() });

describe('EmployeeTreeComponent', () => {
    const Data = [
        { employeeName: "Alan", employeeId: "100", managerId: "150" },
        { employeeName: "Martin", employeeId: "220", managerId: "100" },
        { employeeName: "Jamie", employeeId: "150", managerId: "" },
        { employeeName: "Jimmy", employeeId: "250", managerId: "" },
        { employeeName: "Alex", employeeId: "275", managerId: "100" },
        { employeeName: "Steve", employeeId: "400", managerId: "150" },
        { employeeName: "David", employeeId: "190", managerId: "400" }
      ];
  it('should render only single CEO even when multiple CEO are present', () => {
    const component = shallow(<EmployeeComponent EmployeeData={Data} />);
  
    expect(component).toMatchSnapshot();
  });

  it('should not render duplicate records', () => {
    const Data = [
        { employeeName: "Alan", employeeId: "100", managerId: "150" },
        { employeeName: "Martin", employeeId: "220", managerId: "100" },
        { employeeName: "Martin", employeeId: "220", managerId: "100" },
        { employeeName: "Jimmy", employeeId: "250", managerId: "" },
        { employeeName: "Alex", employeeId: "275", managerId: "100" },
        { employeeName: "Steve", employeeId: "400", managerId: "150" },
        { employeeName: "David", employeeId: "190", managerId: "400" }
      ];
   
    const component = shallow(<EmployeeComponent EmployeeData={Data} />);
  
    expect(component).toMatchSnapshot();
  });
  it('should not only valid employee hierarchy', () => {
    const Data = [
        { employeeName: "Alan", employeeId: "100", managerId: "150" },
        { employeeName: "Martin", employeeId: "220", managerId: "100" },
        { employeeName: "Martin", employeeId: "220", managerId: "100" },
        { employeeName: "Jimmy", employeeId: "250", managerId: "" },
        { employeeName: "Alex", employeeId: "275", managerId: "100" },
        { employeeName: "Steve", employeeId: "400", managerId: "150" },
        { employeeName: "David", employeeId: "190", managerId: "400" }
      ];
   
    const component = shallow(<EmployeeComponent EmployeeData={Data} />);
  
    expect(component).toMatchSnapshot();
  });

  it('should not render tree when no CEO', () => {
    const Data = [
        { employeeName: "Alan", employeeId: "100", managerId: "150" },
        { employeeName: "Martin", employeeId: "220", managerId: "100" },
        { employeeName: "Martin", employeeId: "220", managerId: "100" },
        { employeeName: "Jimmy", employeeId: "250", managerId: "" },
        { employeeName: "Alex", employeeId: "275", managerId: "100" },
        { employeeName: "Steve", employeeId: "400", managerId: "150" },
        { employeeName: "David", employeeId: "190", managerId: "400" }
      ];
   
    const component = shallow(<EmployeeComponent EmployeeData={Data} />);
  
    expect(component).toMatchSnapshot();
  });


});