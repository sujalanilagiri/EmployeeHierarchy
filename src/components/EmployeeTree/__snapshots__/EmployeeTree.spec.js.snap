// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeTreeComponent should not only valid employee hierarchy 1`] = `
<Fragment>
  <div
    className="App"
    id="initechOrgChart"
  >
    <OrgChart
      NodeComponent={[Function]}
      tree={
        Object {
          "value": Object {
            "employeeId": "250",
            "employeeName": "Jimmy",
            "managerId": "",
          },
        }
      }
    />
    <RejectedList
      data={
        Array [
          Object {
            "name": Array [
              "Martin",
            ],
            "reason": "Duplicate Entry of Employee",
          },
          Object {
            "name": Array [
              "100",
            ],
            "reason": "Manager is not a valid Employee",
          },
          Object {
            "name": Array [
              "150",
            ],
            "reason": "Manager is not a valid Employee",
          },
          Object {
            "name": Array [
              "400",
            ],
            "reason": "Manager is not a valid Employee",
          },
        ]
      }
    />
  </div>
</Fragment>
`;

exports[`EmployeeTreeComponent should not render duplicate records 1`] = `
<Fragment>
  <div
    className="App"
    id="initechOrgChart"
  >
    <OrgChart
      NodeComponent={[Function]}
      tree={
        Object {
          "value": Object {
            "employeeId": "250",
            "employeeName": "Jimmy",
            "managerId": "",
          },
        }
      }
    />
    <RejectedList
      data={
        Array [
          Object {
            "name": Array [
              "Martin",
            ],
            "reason": "Duplicate Entry of Employee",
          },
          Object {
            "name": Array [
              "100",
            ],
            "reason": "Manager is not a valid Employee",
          },
          Object {
            "name": Array [
              "150",
            ],
            "reason": "Manager is not a valid Employee",
          },
          Object {
            "name": Array [
              "400",
            ],
            "reason": "Manager is not a valid Employee",
          },
        ]
      }
    />
  </div>
</Fragment>
`;

exports[`EmployeeTreeComponent should not render tree when no CEO 1`] = `
<Fragment>
  <div
    className="App"
    id="initechOrgChart"
  >
    <OrgChart
      NodeComponent={[Function]}
      tree={
        Object {
          "value": Object {
            "employeeId": "250",
            "employeeName": "Jimmy",
            "managerId": "",
          },
        }
      }
    />
    <RejectedList
      data={
        Array [
          Object {
            "name": Array [
              "Martin",
            ],
            "reason": "Duplicate Entry of Employee",
          },
          Object {
            "name": Array [
              "100",
            ],
            "reason": "Manager is not a valid Employee",
          },
          Object {
            "name": Array [
              "150",
            ],
            "reason": "Manager is not a valid Employee",
          },
          Object {
            "name": Array [
              "400",
            ],
            "reason": "Manager is not a valid Employee",
          },
        ]
      }
    />
  </div>
</Fragment>
`;

exports[`EmployeeTreeComponent should render only single CEO even when multiple CEO are present 1`] = `
<Fragment>
  <div
    className="App"
    id="initechOrgChart"
  >
    <OrgChart
      NodeComponent={[Function]}
      tree={
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "value": Object {
                    "employeeId": "220",
                    "employeeName": "Martin",
                    "managerId": "100",
                  },
                },
                Object {
                  "value": Object {
                    "employeeId": "275",
                    "employeeName": "Alex",
                    "managerId": "100",
                  },
                },
              ],
              "value": Object {
                "employeeId": "100",
                "employeeName": "Alan",
                "managerId": "150",
              },
            },
            Object {
              "children": Array [
                Object {
                  "value": Object {
                    "employeeId": "190",
                    "employeeName": "David",
                    "managerId": "400",
                  },
                },
              ],
              "value": Object {
                "employeeId": "400",
                "employeeName": "Steve",
                "managerId": "150",
              },
            },
          ],
          "value": Object {
            "employeeId": "150",
            "employeeName": "Jamie",
            "managerId": "",
          },
        }
      }
    />
    <RejectedList
      data={
        Array [
          Object {
            "name": Array [
              "Jimmy",
            ],
            "reason": "Only on CEO can be shown in the Hierarchy",
          },
        ]
      }
    />
  </div>
</Fragment>
`;
